book/java-rebuilds-service.md

```diff
@@ -0,0 +1,154 @@
+# JVM Build Service
+
+## Overview
+
+The JVM Build Service (JBS) is a controller that will rebuild Java and other JVM Language based dependencies from source.
+
+Java uses a binary distribution model, so the only way to ensure an application is completly built from source is to rebuild all its component libraries from source themselves. Unfortunatly due to the nature of the Java ecosystem this is not an easy process and would normally require a large amount of manual toil.
+
+JBS will attempt to automate this proccess as much as possible, making it much less time consuming to create fully supply chain controlled Java builds.
+
+### Dependencies
+
+JBS depends on Tekton, but is otherwise largely independent of the rest of RHTAP. Given the right configuration it can be deployed outside of RHTAP, which is mainly used for testing and development.
+
+## Architecture
+
+### Flow Overview
+
+The JVM Build service can be enabled on a specific namespace by creating a `JBSConfig` object, with `enableRebuilds: true` specified. This will trigger the controller to create a deployment of the local Cache in the namespace. This cache has a number of purposes, but the primary one is to cache artifacts from Maven central to speed up Java builds.
+
+Once it is enabled and the cache has started the JVM Build Service will watch for `PipelineRun` objects with the `JAVA_COMMUNITY_DEPENDENCIES` results. This will be present on end user builds that have dependencies from untrusted upstream sources.
+
+When one of these `PipelineRun` objects is observed JBS will extra the dependency list, and attempt to rebuild all the listed artifacts from source.
+
+The first stage of this process is to create an `ArtifactBuild` object. This object represents a Maven `group:artifact:version` (GAV) coordinate of the library that needs to be built from source.
+
+Once the `ArtifactBuild` object has been created JBS will then attempt to try and build it. This first step is to attempt to find the relevant source code.
+
+Once the source code location has been discovered a `DependencyBuild` object is created. There are generally less `DependencyBuild` objects than there are `ArtifactBuild`, as multiple artifacts can come from the same build. The controller will then try and build the artifact, first it will run a build discovery pipeline, that attempts to determine possible ways of building the artifact. Once discovery is complete it uses this information to attempt to build the artifact in a trial and error manner.
+
+If a build is successful the results are stored in a container image and the state is marked as complete, otherwise it is marked as failed and manual effort is required to fix the build.
```

Comment:
Are there common types of "manual effort" that may be needed?

Reply:
Yes, but that is really an end user concern: https://github.com/redhat-appstudio/docs.appstudio.io/blob/main/docs/modules/ROOT/pages/how-to-guides/Secure-your-supply-chain/proc_java_dependencies.adoc#dealing-with-failed-builds

How much overlap do we want to have between these docs?

Reply:
If you don't want to duplicate information between the documents, can you link to that document from this one?